llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -5798,17 +5798,25 @@ InstructionCost AArch64TTIImpl::getPartialReductionCost(
       (!ST->isNeonAvailable() || !ST->hasDotProd()))
     return Invalid;
 
-  if ((Opcode != Instruction::Add && Opcode != Instruction::Sub) ||
+  if ((Opcode != Instruction::Add && Opcode != Instruction::Sub &&
+       Opcode != Instruction::FAdd) ||
       OpAExtend == TTI::PR_None)
     return Invalid;
 
+  // Floating-point partial reductions are invalid if `reassoc` and `contract`
+  // are not allowed.
+  if (Opcode == Instruction::FAdd &&
+      (!FMF->allowReassoc() || !FMF->allowContract()))
```

Comment:
This should not crash if `FMF` is null (it should just return an invalid cost)
```suggestion
      (!FMF ||  !FMF->allowReassoc() || !FMF->allowContract()))
```

Reply:
Sure, it would be good to have an assert added as well.